### PR TITLE
Refactor & Improvement

### PR DESCRIPTION
**Pull request by kazutoiris**
1. Modified `injectFunctions` for improved simplicity and generality.

2. Updated compatibility for IDE versions supported by the plugin.

3. Introduced a new plugin icon.

4. Added contributor.

Here's the compiled plugin after these changes: [chiselinstantiate-0.0.1-SNAPSHOT.zip](https://github.com/user-attachments/files/16993487/chiselinstantiate-0.0.1-SNAPSHOT.zip)

### CODE REVIEW

File: idealchisel/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <idea-plugin>
   <id>org.chipsalliance.idealchisel</id>
   <name>Chisel Support</name>
-  <vendor>Tesla Zhang, Jiuyang Liu</vendor>
+  <vendor>Tesla Zhang, Jiuyang Liu, kazutoiris</vendor>
```
Comment (sequencer) [inline idealchisel/resources/META-INF/plugin.xml]: Please provide a passport name.

File: idealchisel/src/InstanceInjector.scala [hunk 2]
```diff
@@ -6,22 +6,14 @@ import org.jetbrains.plugins.scala.lang.psi.api.toplevel.typedef.{ScObject, ScTy
...
-        )
-        .toSeq
-    case _ => Nil
+  // add an implicit converter from Instance[T] to T, then IDEA will be happy
+  override def injectFunctions(source: ScTypeDefinition): Seq[String] = {
+    source match {
+      case o: ScObjectImpl if o.isPackageObject && targetObject.equals(o.qualifiedName) =>
+        Seq("implicit def from[T](instance: Instance[T]): T = ???")
+      case _ => Seq.empty
+    }
```
Comment (sequencer) [inline idealchisel/src/InstanceInjector.scala:17]:
```
✦ ❯ mill idealchisel.release
[42/44] idealchisel.compile
[info] compiling 1 Scala source to /home/sequencer/projects/idealchisel/out/idealchisel/compile.dest/classes ...
[error] /home/sequencer/projects/idealchisel/idealchisel/src/InstanceInjector.scala:14:15: not found: type ScObjectImpl
[error]       case o: ScObjectImpl if o.isPackageObject && targetObject.equals(o.qualifiedName) =>
[error]               ^
[error] one error found
1 targets failed
idealchisel.compile Compilation failed
```

**Review by sequencer — CHANGES_REQUESTED** [review]

File: idealchisel/src/InstanceInjector.scala (hunk 2 above)
Comment (kazutoiris) [reply idealchisel/src/InstanceInjector.scala:17]: Fixed in the latest commit.

File: idealchisel/resources/META-INF/plugin.xml (hunk 1 above)
Comment (kazutoiris) [reply idealchisel/resources/META-INF/plugin.xml]: modified in the latest commit.

**Pull request merged**